txnkv/txnsnapshot/snapshot.go

```diff
@@ -255,7 +255,8 @@ func (s *KVSnapshot) BatchGetWithTier(ctx context.Context, keys [][]byte, readTi
 	// Create a map to collect key-values from region servers.
 	var mu sync.Mutex
 	err := s.batchGetKeysByRegions(bo, keys, readTier, func(k, v []byte) {
-		if len(v) == 0 {
+		// when read buffer tier, empty value means a delete record, should also collect it.
```

Comment:
This is confusing, is it an extra fix?

Reply:
It's the different behaviors between snapshot read and buffer read. For a delete record, buffer read return empty value, e.g.:

```go
memdb := newMemDB()
memdb.Delete([]byte("k1"))
v, err := memdb.Get([]byte("k1")) // v == []byte{}, err == nil
``` 

The related [TiKV PR](https://github.com/tikv/tikv/pull/16646).